components/greaselion/browser/greaselion_converted_extension.h

```diff
@@ -0,0 +1,42 @@
+/* Copyright 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_COMPONENTS_GREASELION_BROWSER_GREASELION_CONVERTED_EXTENSION_H_
+#define BRAVE_COMPONENTS_GREASELION_BROWSER_GREASELION_CONVERTED_EXTENSION_H_
+
+#include <memory>
+#include <utility>
+
+#include "base/files/scoped_temp_dir.h"
+#include "base/macros.h"
+#include "base/memory/weak_ptr.h"
+
+namespace extensions {
+class Extension;
+}  // namespace extensions
+
+namespace greaselion {
+
+class GreaselionConvertedExtension {
+ public:
+  GreaselionConvertedExtension(
+      scoped_refptr<extensions::Extension> extension,
+      std::unique_ptr<base::ScopedTempDir> extension_dir);
+  ~GreaselionConvertedExtension();
+
+  scoped_refptr<extensions::Extension> extension() const { return extension_; }
+  std::unique_ptr<base::ScopedTempDir> extension_dir() {
+    return std::move(extension_dir_);
```

Comment:
Style-wise, probably don't want to modify the object with a simple `accessor()`-like name.

Maybe return a `std::unique_ptr<base::ScopedTempDir>&` and let the caller move it?

Reply:
Good point, fixed!